### DISCUSSION TITLE
Hox to fix `#include errors detected`?

### DISCUSSION THREAD

**Discussion by NhanAZ** [microsoft/vscode-cpptools#9641] [Q&A] [answered]
Here is my code
```cpp
#include<iostream>

using namespace std;

int main() {
	cout << "Hello World!";
}
```
Here is the error message
```
#include errors detected. Please update your includePath. Squiggles are disabled for this translation unit (C:\Users\NhanAZ\Downloads\CPP environment\cpp.cpp).
```
```
cannot open source file "iostream"
```
```
[{
	"resource": "/C:/Users/NhanAZ/Downloads/CPP environment/cpp.cpp",
	"owner": "C/C++",
	"code": "1696",
	"severity": 8,
	"message": "#include errors detected. Please update your includePath. Squiggles are disabled for this translation unit (C:\\Users\\NhanAZ\\Downloads\\CPP environment\\cpp.cpp).",
	"source": "C/C++",
	"startLineNumber": 1,
	"startColumn": 1,
	"endLineNumber": 1,
	"endColumn": 19
},{
	"resource": "/C:/Users/NhanAZ/Downloads/CPP environment/cpp.cpp",
	"owner": "C/C++",
	"code": "1696",
	"severity": 8,
	"message": "cannot open source file \"iostream\"",
	"source": "C/C++",
	"startLineNumber": 1,
	"startColumn": 1,
	"endLineNumber": 1,
	"endColumn": 19
}]
```

**Comment by sean-mcmanus** [comment]
What is your compilerPath set to? What do you get when you run the C/C++: Log Diagnostics command from the Command Palette after opening the cpp.cpp file?

**Reply by NhanAZ** [reply]
```
-------- Diagnostics - 7/27/2022, 1:45:29 AM
Version: 1.11.4
Current Configuration:
{
    "name": "Win32",
    "includePath": [
        "${workspaceFolder}/**"
    ],
    "defines": [
        "_DEBUG",
        "UNICODE",
        "_UNICODE"
    ],
    "cStandard": "c17",
    "cppStandard": "c++17",
    "intelliSenseMode": "windows-msvc-x64",
    "intelliSenseModeIsExplicit": false,
    "cStandardIsExplicit": false,
    "cppStandardIsExplicit": false,
    "mergeConfigurations": false,
    "compilerPathIsExplicit": false,
    "browse": {
        "path": [
            "${workspaceFolder}/**"
        ],
        "limitSymbolsToIncludedHeaders": true
    }
}
No active translation units.
```

**Reply by sean-mcmanus** [reply]
It appears you have no compilerPath set. Do you have a compiler installed?

**Reply by Sa458** [reply]
no

**Comment by michelleangela** [comment]
@NhanAZ it looks like the extension cannot locate the file iostream. The extension will get the path of system libraries based on an installed compiler's path.

Could you run the command `C/C++: Log Diagnostics` from the VS Code command palette. This will print out your IntelliSense configuration which will show the compiler path configuration.

**Comment by michelleangela** [comment, marked as answer]
Also, for additional information about IntelliSense configuration based on your platform, you can go to https://code.visualstudio.com/docs/languages/cpp#_tutorials

**Reply by NhanAZ** [reply]
The documentation is of no help to me, it seems rather confusing to newbies.
Do you have a video tutorial? It will be easier to understand than a dry document.

**Reply by michelleangela** [reply]
> The documentation is of no help to me, it seems rather confusing to newbies. Do you have a video tutorial? It will be easier to understand than a dry document.

We do not have video tutorials.

From your logs, it looks like you are on Windows. To install a compiler on Windows, follow step number 3 of https://code.visualstudio.com/docs/cpp/config-msvc#_prerequisites. When you have a compiler installed, the extension should be able to find the compiler automatically.
